nsqd/diskqueue.go

```diff
@@ -341,6 +353,10 @@ func (d *diskQueue) writeOne(data []byte) error {
 	d.writePos += totalBytes
 	atomic.AddInt64(&d.depth, 1)
 
+	// TODO: this is off by one, but we cannot fix this without also fixing
+	// readOne above. We cannot fix readOne above until we guarantee that
+	// every old diskqueue format has been fully processed, otherwise we
+	// will miss one message from old formatted files.
```

Comment:
This is meant to rotate files when we reach maxBytesPerFile, but will only
rotate when we exceed that number. If max bytes is 1024 and we are at bytes
1024, we will write one more message before rotating files.

On Saturday, July 11, 2015, Matt Reiferson <notifications@github.com> wrote:

> In nsqd/diskqueue.go
> <https://github.com/bitly/nsq/pull/600#discussion_r34415803>:
>
> > @@ -341,6 +353,10 @@ func (d *diskQueue) writeOne(data []byte) error {
> >  	d.writePos += totalBytes
> >  	atomic.AddInt64(&d.depth, 1)
> >
> > +	// TODO: this is off by one, but we cannot fix this without also fixing
> > +	// readOne above. We cannot fix readOne above until we guarantee that
> > +	// every old diskqueue format has been fully processed, otherwise we
> > +	// will miss one message from old formatted files.
>
> what do you mean by this comment?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitly/nsq/pull/600/files#r34415803>.
>